src/openforms/formio/formatters/formio.py

```diff
@@ -166,50 +80,44 @@ def process_result(self, component: Component, formatted: str) -> str:
             return _("attachment: %s") % formatted
         return formatted
 
-    def format(self, component: Component, value: List) -> str:
+    def format(self, component: Component, value: dict) -> str:
         # this is only valid for display to the user (because filename component option, dedupe etc)
         return value["originalName"]
 
 
-@register("textarea")
-class TextAreaFormatter(FormioFormatter):
+class TextAreaFormatter(FormatterBase):
     def format(self, component: Component, value: str) -> str:
         # TODO custom formatting?
         return str(value)
 
 
-@register("number")
-class NumberFormatter(FormioFormatter):
-    def format(self, component: Component, value: Union[int, float]) -> str:
+class NumberFormatter(FormatterBase):
+    def format(self, component: Component, value: int | float) -> str:
         # localized and forced to decimalLimit
         return number_format(value, decimal_pos=component.get("decimalLimit"))
 
 
-@register("password")
-class PasswordFormatter(FormioFormatter):
+class PasswordFormatter(FormatterBase):
     def format(self, component: Component, value: str) -> str:
         # TODO legacy just printed as-is, but we might want to use unicode-dots or stars
         # return "\u25CF" * len(value)
         return str(value)
 
 
-@register("checkbox")
-class CheckboxFormatter(FormioFormatter):
+class CheckboxFormatter(FormatterBase):
     def format(self, component: Component, value: bool) -> str:
-        return yesno(value)
+        return str(yesno(value))
```

Comment:
that also threw me off, my editor was warning me about the mismatching type hints

it can happen if you call `yesno` with an optional arg: `yesno(value, "yep,nope,maybe")` where the optional arg is not 2 or 3 items, then the unmodified `value` is returned which can be any type.

It's really a change to shut up the type checker tbh.